### PR TITLE
(feat)(Business Glossary) add tabular schema and new UI for business glossary

### PR DESCRIPTION
**Pull request by saxo-lalrishav**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

image
image

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java [hunk 1]
```diff
@@ -909,4 +911,12 @@ private void configureGlossaryRelationshipResolvers(final RuntimeWiring.Builder
...
         }), loaderOptions);
     }
+    private void configureGlossaryTermResolvers(final RuntimeWiring.Builder builder) {
+        builder
+                .type("GlossaryTerm", typeWiring -> typeWiring
+                        .dataFetcher("schemaMetadata", new AuthenticatedResolver<>(
+                                new AspectResolver())
+                        )
+                );
+    }
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java:921]: lets put this method near the other configureXyzResolvers

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java [hunk 2]
```diff
@@ -137,4 +150,62 @@ public BrowseResults browse(@Nonnull List<String> path,
...
+
+        final ConjunctivePrivilegeGroup allPrivilegesGroup = new ConjunctivePrivilegeGroup(ImmutableList.of(
+                PoliciesConfig.EDIT_ENTITY_PRIVILEGE.getType()
+        ));
+
+        List<String> specificPrivileges = new ArrayList<>();
+
+        if (updateInput.getEditableSchemaMetadata() != null) {
+            specificPrivileges.add(PoliciesConfig.EDIT_GLOSSARY_TERM_COL_TAGS_PRIVILEGE.getType());
+            specificPrivileges.add(PoliciesConfig.EDIT_GLOSSARY_TERM_COL_DESCRIPTION_PRIVILEGE.getType());
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java:199]: are we sure we want to support tagging of glossary terms schemas? This seems like it could lead to schema tagging inception, since schemas can be tagged with terms which can then have their own schemas with their own tags- is this going to get confusing?

cc @shirshanka

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java [hunk 3]
```diff
@@ -4,17 +4,23 @@
...
+import com.linkedin.schema.SchemaMetadata;
+
+
 import javax.annotation.Nonnull;
 
 
 /**
  * Maps Pegasus {@link RecordTemplate} objects to objects conforming to the GQL schema.
- *
+ * <p>
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java:23]: nit- accidental

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermUpdateInputSnapshotMapper.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        output.setFieldPath(schemaFieldInfo.getFieldPath());
+
+        if (schemaFieldInfo.getGlobalTags() != null) {
+            final GlobalTags globalTags = new GlobalTags();
+            globalTags.setTags(new TagAssociationArray(schemaFieldInfo.getGlobalTags().getTags().stream().map(
+                    element -> TagAssociationUpdateMapper.map(element)).collect(Collectors.toList())));
+            output.setGlobalTags(globalTags);
+        }
+        return output;
+    }
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermUpdateInputSnapshotMapper.java:84]: shouldn't this be a shared util? why do we need to rewrite this

File: datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx [hunk 5]
```diff
@@ -45,7 +53,51 @@ export class GlossaryTermEntity implements Entity<GlossaryTerm> {
...
+                        name: 'Related Entities',
+                        component: GlossaryRelatedEntity,
+                    },
+                    {
+                        name: 'Related Terms',
+                        component: GlossayRelatedTerms,
+                    },
+                    {
+                        name: 'Properties',
+                        component: GlossaryTermPropertyTab,
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx:78]: does this need to be GlossaryTerm specific? I think we can use the general PropertyTab right?

File: datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedEntity.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            const entities = entitySearchResult[type].data?.search?.searchResults;
+            if (entities && entities.length > 0) {
+                filteredSearchResult[type] = entities;
+            }
+        });
+
+        return filteredSearchResult;
+    }, [entitySearchResult]);
+
+    return <RelatedEntityResults searchResult={entitySearchForDetails} />;
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedEntity.tsx:37]: from the screenshot, it looks like the spacing on this section is a little off on the glossary page. the entity result seems more indented from the title in a way that looks a bit strange?

File: datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedTerms.tsx [hunk 7]
```diff
@@ -26,7 +23,8 @@ const Content = styled.div`
     flex-grow: 1;
 `;
 
-export default function GlossayRelatedTerms({ glossaryTerm }: Props) {
+export default function GlossayRelatedTerms() {
+    const { entityData }: any = useEntityData();
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedTerms.tsx:27]: i dont think this `: any` type annotation helps at all?

File: datahub-web-react/src/graphql/glossaryTerm.graphql [hunk 8]
```diff
@@ -42,5 +48,45 @@ query getGlossaryTerm($urn: String!, $start: Int, $count: Int) {
...
+        }
+        editableSchemaMetadata {
+            editableSchemaFieldInfo {
+                fieldPath
+                description
+                globalTags {
+                    ...globalTagsFields
+                }
+            }
+        }
```
Comment (gabe-lyons) [inline datahub-web-react/src/graphql/glossaryTerm.graphql:90]: is this necessary? we have turned off schema version history and still have some work to do before we re-enable it. Can we get away w/ not supporting this for now?

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl [hunk 9]
```diff
@@ -47,9 +47,4 @@ record GlossaryTermInfo {
    * A key-value map to capture any other non-standardized properties for the glossary term
    */
   customProperties: map[string, string] = { }
-
-  /**
-   * Schema definition of the glossary term
-   */
-  rawSchema: optional string
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl:54]: we cannot delete fields on models. this is a breaking change. Use the deprecated annotation instead!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java (hunk 1 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java]: done

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java (hunk 2 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java]: removed

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermUpdateInputSnapshotMapper.java (hunk 4 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermUpdateInputSnapshotMapper.java]: removed

File: datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedEntity.tsx (hunk 6 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedEntity.tsx:37]: fixed and updated screenshot

File: datahub-web-react/src/graphql/glossaryTerm.graphql (hunk 8 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/graphql/glossaryTerm.graphql]: removed

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java (hunk 3 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java]: deleted

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl (hunk 9 above)
Comment (saxo-lalrishav) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryTermInfo.pdl]: done

File: datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx (hunk 5 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx:80]: It differs in https://github.com/linkedin/datahub/blob/34c27f076bb7c4c16a295292cee17a7d07fe6165/datahub-web-react/src/app/entity/shared/tabs/Properties/PropertiesTab.tsx#L49 and https://github.com/saxo-datagov/datahub/blob/62d3b692df21b3a748c0f1012bf7ea6bb65bf093/datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryTermPropertyTab.tsx#L48

File: datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedTerms.tsx (hunk 7 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/glossaryTerm/profile/GlossaryRelatedTerms.tsx]: agree!!, removed

File: datahub-graphql-core/src/main/resources/entity.graphql [hunk 10]
```diff
@@ -115,6 +115,7 @@ type Mutation {
     """
     updateDataset(urn: String!, input: DatasetUpdateInput!): Dataset
 
+
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/resources/entity.graphql:118]: spurious?

File: datahub-graphql-core/src/main/resources/entity.graphql [hunk 11]
```diff
@@ -749,6 +750,18 @@ type GlossaryTerm implements Entity {
     Edges extending from this entity
     """
     relationships(input: RelationshipsInput!): EntityRelationshipsResult
+
+    
+    """
+    Schema metadata of the dataset
+    """
+    schema: Schema @deprecated(reason: "Use `schemaMetadata`")
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/resources/entity.graphql:758]: lets not add the deprecated one- schemaMetadata is fine. lets remove this line

File: datahub-graphql-core/src/main/resources/entity.graphql [hunk 12]
```diff
@@ -2363,6 +2376,7 @@ input DatasetUpdateInput {
     editableProperties: DatasetEditablePropertiesUpdate
 }
 
+
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/resources/entity.graphql:2379]: spurious?

File: datahub-web-react/datahub-frontend.graphql [hunk 13]
```diff
@@ -24,6 +24,7 @@ input DatasetUpdateInput {
     ownership: OwnershipUpdate
 }
 
+
```
Comment (gabe-lyons) [inline datahub-web-react/datahub-frontend.graphql:27]: spurious?

File: datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx (hunk 5 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx:80]: Got it- we should use `getOverridePropertiesFromEntity` to patch this up then. See how we do this for externalUrl for datasets:

https://github.com/linkedin/datahub/blob/master/datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx#L177

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTable.tsx [hunk 14]
```diff
@@ -55,7 +55,7 @@ export default function SchemaTable({
     const [selectedFkFieldPath, setSelectedFkFieldPath] =
         useState<null | { fieldPath: string; constraint?: ForeignKeyConstraint | null }>(null);
 
-    const descriptionRender = useDescriptionRenderer(editableSchemaMetadata);
+    const descriptionRender = useDescriptionRenderer(editableSchemaMetadata, editMode);
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTable.tsx:58]: i would prefer to pass editMode in via react context so we dont have to pass the param all around. Something like

`SchemaEditableContext`?

File: datahub-web-react/src/app/entity/shared/types.ts [hunk 15]
```diff
@@ -23,11 +23,12 @@ import { FetchedEntity } from '../../lineage/types';
 
 export type EntityTab = {
     name: string;
-    component: React.FunctionComponent;
+    component: React.FunctionComponent<{ properties?: any }>;
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/types.ts:26]: properties should not be `any`. Let's make the type explicit.

File: datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx [hunk 16]
```diff
@@ -42,10 +38,10 @@ export default ({ searchResult, entityPath }: Props) => {
                 dataSource={entitiesToShow}
                 renderItem={(item) => {
                     return (
-                        <ListItem>
+                        <>
                             {entityRegistry.renderPreview(entityType, PreviewType.PREVIEW, item.entity)}
                             <Divider />
-                        </ListItem>
+                        </>
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx:44]: how sure are we that this won't affect other screens? Could this mess w/ the formatting of list items in search, browse, user profile page, etc?

File: datahub-web-react/src/graphql/glossaryTerm.graphql [hunk 17]
```diff
@@ -1,3 +1,5 @@
+
+
```
Comment (gabe-lyons) [inline datahub-web-react/src/graphql/glossaryTerm.graphql:2]: spurious

File: datahub-web-react/src/app/entity/shared/types.ts (hunk 15 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/types.ts:26]: but different tabs may have different property , isnt it ?

File: datahub-web-react/src/graphql/glossaryTerm.graphql [hunk 18]
```diff
@@ -42,5 +44,13 @@ query getGlossaryTerm($urn: String!, $start: Int, $count: Int) {
                 value
             }
         }
+         schemaMetadata(version: 0) {
+            ...schemaMetadataFields
+        }
+        previousSchemaMetadata: schemaMetadata(version: -1) {
+            ...schemaMetadataFields
+        }
```
Comment (gabe-lyons) [inline datahub-web-react/src/graphql/glossaryTerm.graphql:52]: we don't need this since we aren't displaying schema version history

File: datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx (hunk 16 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/shared/entitySearch/RelatedEntity.tsx:44]: image

this are the component which are using RelatedEntity.ts directly or indirectly
Verified for all of them

File: datahub-graphql-core/src/main/resources/entity.graphql (hunk 11 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/resources/entity.graphql]: deleted

File: datahub-graphql-core/src/main/resources/entity.graphql (hunk 12 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/resources/entity.graphql]: deleted

File: datahub-web-react/datahub-frontend.graphql (hunk 13 above)
Comment (saxo-lalrishav) [reply datahub-web-react/datahub-frontend.graphql]: deleted

File: datahub-web-react/src/graphql/glossaryTerm.graphql (hunk 18 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/graphql/glossaryTerm.graphql]: removed

File: datahub-web-react/src/graphql/glossaryTerm.graphql (hunk 17 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/graphql/glossaryTerm.graphql]: removed

File: datahub-web-react/src/app/entity/shared/types.ts (hunk 15 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/types.ts]: added new type to properties

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTable.tsx (hunk 14 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/SchemaTable.tsx]: yes, done

File: datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx (hunk 5 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx]: yes tanks for ref, done

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx [hunk 19]
```diff
@@ -32,7 +32,6 @@ export default function useDescriptionRenderer(editableSchemaMetadata: EditableS
                         },
                     }).then(refetch)
                 }
-                editable
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx:35]: wouldn't this make dataset schemas not editable? I think we need to pass this in via context or else we wont be able to share this logic btwn datasets & glossary terms

File: datahub-web-react/src/app/entity/shared/types.ts [hunk 20]
```diff
@@ -21,13 +21,18 @@ import {
...
+
 export type EntityTab = {
     name: string;
-    component: React.FunctionComponent;
+    component: React.FunctionComponent<{ properties?: GenericEntityTabProperties }>;
     display?: {
         visible: (GenericEntityProperties, T) => boolean; // Whether the tab is visible on the UI. Defaults to true.
         enabled: (GenericEntityProperties, T) => boolean; // Whether the tab is enabled on the UI. Defaults to true.
     };
+    properties?: any;
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/types.ts:35]: properties should not be any

**Review by gabe-lyons — COMMENTED** [review]
looking good- just a few things to clean up

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx (hunk 19 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx:35]: https://github.com/linkedin/datahub/pull/3813/files#diff-a628caf403fe8ddae8c9934ae260353cd62c9c4613bab82f90ce9b7fe0d04ffa
i have already added a context for this

Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx:35]: will update this component with context

Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx:35]: Sorry, i am little bit confused
I have already used ` const isSchemaEditable = React.useContext(SchemaEditableContext);`  in `SchemaDescriptionField.tsx`
and used the `<SchemaEditableContext.Provider>` in `SchemaTab.tsx`
wouldnt that be enough to share the editing logic between datasets and glossary terms , or i need to do some more modification ?

File: datahub-web-react/src/app/entity/shared/types.ts (hunk 20 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/types.ts]: updated

File: datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx (hunk 19 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/entity/shared/tabs/Dataset/Schema/utils/useDescriptionRenderer.tsx:35]: oh I see- I was confused. Thanks!

**Review by gabe-lyons — APPROVED** [review]
looks good

File: datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx [hunk 21]
```diff
@@ -45,7 +53,55 @@ export class GlossaryTermEntity implements Entity<GlossaryTerm> {
...
+                entityType={EntityType.GlossaryTerm}
+                useEntityQuery={useGetGlossaryTermQuery as any}
+                tabs={[
+                    {
+                        name: 'Schema',
+                        component: SchemaTab,
+                        properties: {
+                            editMode: false,
+                        },
+                    },
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx:69]: one last comment- we should add logic to make this tab not visible in the case there is no schema, similar to here:


https://github.com/linkedin/datahub/blob/master/datahub-web-react/src/app/entity/dataset/DatasetEntity.tsx#L93

Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/glossaryTerm/GlossaryTermEntity.tsx]: added

**Review by gabe-lyons — APPROVED** [review]
lgtm!

File: metadata-models/src/main/pegasus/com/linkedin/metadata/aspect/GlossaryTermAspect.pdl [hunk 22]
```diff
@@ -6,8 +6,9 @@ import com.linkedin.common.Ownership
...
 import com.linkedin.common.BrowsePaths
 import com.linkedin.glossary.GlossaryRelatedTerms
+import com.linkedin.schema.SchemaMetadata
 
 /**
  * A union of all supported metadata aspects for a CorpUser
  */
-typeref GlossaryTermAspect = union[GlossaryTermKey, GlossaryTermInfo, Ownership, Status, BrowsePaths, GlossaryRelatedTerms]
\ No newline at end of file
+typeref GlossaryTermAspect = union[GlossaryTermKey, GlossaryTermInfo, Ownership, Status, BrowsePaths, GlossaryRelatedTerms, SchemaMetadata]
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/metadata/aspect/GlossaryTermAspect.pdl:14]: can you revert this change? we no longer need to extend the GlossaryTermAspect pdl file

Comment (saxo-lalrishav) [reply metadata-models/src/main/pegasus/com/linkedin/metadata/aspect/GlossaryTermAspect.pdl]: updated

File: datahub-web-react/src/app/entity/shared/types.ts [hunk 23]
```diff
@@ -21,13 +21,18 @@ import {
 } from '../../../types.generated';
 import { FetchedEntity } from '../../lineage/types';
 
+export type GenericEntityTabProperties = {
+    editMode: boolean;
+};
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/types.ts:26]: thinking about it again, you were right to have typed this as `any`. Let's go back to that.

**Review by gabe-lyons — COMMENTED** [review]
looks good aside from my last comment (apologies, I had asked for this in the first place)

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java [hunk 24]
```diff
@@ -61,27 +69,24 @@ public EntityType type() {
 
     @Override
     public List<DataFetcherResult<GlossaryTerm>> batchLoad(final List<String> urns, final QueryContext context) {
-        final List<GlossaryTermUrn> glossaryTermUrns = urns.stream()
-                .map(GlossaryTermUtils::getGlossaryTermUrn)
-                .collect(Collectors.toList());
-
+        final Set<Urn> glossaryTermUrns = urns.stream()
```
Comment (shirshanka) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java:72]: Hi there! This cannot be a set - it must be a list! 

This is because the ordering must be exactly the same as the list of urns passed as an argument

Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java:72]: Hi there! This cannot be a set - it must be a list! 

This is because the ordering must be exactly the same as the list of urns passed as an argument

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java [hunk 25]
```diff
@@ -17,27 +19,29 @@
  *
  * To be replaced by auto-generated mappers implementations
  */
-public class GlossaryTermSnapshotMapper implements ModelMapper<GlossaryTermSnapshot, GlossaryTerm> {
+public class GlossaryTermSnapshotMapper implements ModelMapper<EntityResponse, GlossaryTerm> {
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java:22]: You can rename this to GlossaryTermMapper, now that there's no snapshot involved!

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java (hunk 24 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java:72]: but the function batchGetV2() is taking Set of urn as its parameter
 https://github.com/linkedin/datahub/blob/83d6eef35d3242fdb544955965a9d92fe577f097/metadata-service/restli-client/src/main/java/com/linkedin/entity/client/EntityClient.java#L37

File: datahub-web-react/src/app/entity/shared/types.ts (hunk 23 above)
Comment (saxo-lalrishav) [reply datahub-web-react/src/app/entity/shared/types.ts]: done

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java (hunk 25 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java:22]: sure will do

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java (hunk 24 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/GlossaryTermType.java]: sure updated, as per discussed

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java (hunk 25 above)
Comment (saxo-lalrishav) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/glossary/mappers/GlossaryTermSnapshotMapper.java]: done

**Review by shirshanka — APPROVED** [review]
LGTM!
